### PR TITLE
Deposits under minimum

### PR DESCRIPTION
**Pull request by Valdorff**

### CODE REVIEW

File: RPIPs/RPIP-28.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+discussions-to: 
+status: Draft
+type: Protocol
+category: Core
+created: 2023-08-11
+---
+
+## Abstract
+
+Right now, if an NO falls below the minimum (value of RPL staked <10% of borrowed ETH), they can't
```
Comment (xrchz) [inline RPIPs/RPIP-28.md:15]: typo ("an NO" -> "a NO")

Comment (xrchz) [reply RPIPs/RPIP-28.md:15]: unless this is because you pronounce it en oh?

File: RPIPs/RPIP-28.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+category: Core
+created: 2023-08-11
+---
+
+## Abstract
+
+Right now, if an NO falls below the minimum (value of RPL staked <10% of borrowed ETH), they can't 
+add a new minipool to that node unless they stake enough to reach the minimum AND cover the new
+minipool. Unless, of course, they make a new node. This means that we're creating an incentive to
+make Sybil nodes.
```
Comment (xrchz) [inline RPIPs/RPIP-28.md:18]: side point: I would love if we could find alternative terminology than Sybil -- I realise it's an industry-wide thing, but it's loaded with stigma against mental illness that I cringe at participating in.

Comment (xrchz) [reply RPIPs/RPIP-28.md:18]: "duplicate" would work for this RPIP although it doesn't capture the whole meaning maybe

File: RPIPs/RPIP-28.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+This proposal suggests adding the ability to do a combined minipool deposit AND RPL stake. This
+would allow creating a new minipool even when under the minimum and remove the incentive to create
+Sybil nodes. Because creating a Sybil node is an inconvenience, we can require _slightly_ more RPL
+to be staked, such that the NO moves towards the minimum overall.
+
+## Specification
+
+- There SHALL be a function that allows an atomic RPL stake and minipool ETH deposit
+- This function SHALL succeed if the staked RPL meets of the amount that would be needed for this
```
Comment (xrchz) [inline RPIPs/RPIP-28.md:28]: extra word "meets [of] the"

File: RPIPs/RPIP-28.md (hunk 2 above)
Comment (xrchz) [reply RPIPs/RPIP-28.md:18]: or "sock puppet node"

File: RPIPs/RPIP-28.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    the minimum needed (eg, for RPL rewards) 
+
+## Reference Implementation
+
+- Add a new `RocketNodeDeposit.deposit_with_RPL` function that has the same arguments as
+  `RocketNodeDeposit.deposit` plus a new `rpl_to_stake` argument 
+- `RocketNodeDeposit.deposit` now calls `RocketNodeDeposit.deposit_with_RPL` with rpl_to_stake=0
+- In `RocketNodeDeposit.deposit_with_RPL`
+  - `require(msg.value == _bondAmount, "Invalid value");`
+  - `rpl_requirement_met == (rocketNetworkPrices.getRPLPrice() * rpl_to_stake > rocketDAOProtocolSettingsNode.getMinimumPerMinipoolStake() * (32-_bondAmount))`
```
Comment (xrchz) [inline RPIPs/RPIP-28.md:40]: I think you want an assignment (=) rather than equality test (==) here?

File: RPIPs/RPIP-28.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  - Pass along rpl_requirement_met as needed such that the deposit will succeed if EITHER:
+    - `rpl_requirement_met` is true
```
Comment (xrchz) [inline RPIPs/RPIP-28.md:42]: maybe call it `this_deposit_enough_rpl` or something, to make it clear that it's only one disjunct of the overall RPL requirement?

**Review by xrchz — APPROVED** [review]
LGTM after addressing all my comments

File: RPIPs/RPIP-28.md (hunk 1 above)
Comment (Valdorff) [reply RPIPs/RPIP-28.md:15]: I do pronounce it "en oh" 😛 

Language is silly in that this matters, but here we are

File: RPIPs/RPIP-28.md (hunk 2 above)
Comment (Valdorff) [reply RPIPs/RPIP-28.md:18]: sock puppet works for me

File: RPIPs/RPIP-28.md (hunk 1 above)
Comment (xrchz) [reply RPIPs/RPIP-28.md:15]: either article works for me :)

**Comment by Valdorff** [comment]
@xrchz Fixed all of those issues, and got rid of a sentence in abstract referencing the earlier 110% variant.

**Review by xrchz — APPROVED** [review]

**Pull request merged**